qldpc/abstract.py

```diff
@@ -933,53 +933,102 @@ def iter_mats(cls, dimension: int, field: int | None = None) -> Iterator[galois.
 
 
 class ProjectiveSpecialLinearGroup(Group):
-    """Projective variant of the special linear group (PSL)."""
+    """Projective variant of the special linear group (PSL = SL/center)."""
 
-    # TODO: support linear representation, similarly to SL.
-    # This should straightforward to implement (with a similar construction to that in SL) if we
-    # find a representation of generating matrices of dimension > 2.
 
     _dimension: int
 
-    def __init__(self, dimension: int, field: int | None = None) -> None:
+    def __init__(self, dimension: int, field: int | None = None, linear_rep: bool = True) -> None:
         self._name = f"PSL({dimension},{field})"
         self._dimension = dimension
         self._field = galois.GF(field or DEFAULT_FIELD_ORDER)
-        group: Group
-        if self.field.order == 2:
-            group = SpecialLinearGroup(dimension, 2)
-        elif dimension == 2:
-            group = Group.from_generating_mats(*self.get_generator_mats())
+        
+        if linear_rep:
+            # Construct a linear representation of this group, in which group elements permute
+            # elements of the vector space that the generating matrices act on.
+
+            #root generates the cyclic subgroup of roots of unity of order 'dimension' of the multiplicative group of self.field.
+            num_roots = galois.gcd(self.dimension, self.field.order - 1 )
+            root = self.field.primitive_element ** ( ( self.field.order - 1 ) // num_roots )
+            roots = [ root ** k for k in range(num_roots) ]
+            
+            # PSL acts on the quotient of the target space of SL by roots of unity of order 'dimension'.
```

Comment:
@michaelontiveros can you help me understand why we use roots of unity here, rather than all non-zero scalars (1, 2, ..., `self.field.order`)?

PSL should be SL mod scalars, no?  I guess I'm missing what that implies for the target space of PSL (or rather, I'm missing why it implies that the target space of PSL is equal to [target space of SL] mod [roots of unity of order `dimension`]).

Reply:
Also, I cleaned up this contribution on the `PSL` branch.  Do you want to open a new PR from `PSL` to `main` to have an "official" contribution into `main`? :slightly_smiling_face: 

I would also a appreciate a quick review (if you have the time) to make sure that I did not make any breaking modifications to your code.

Reply:
We mod out by roots of unity of order `dimension` because the scalar matrix `rI` is in the group `SL(dimension)` if and only if the determinant is `1`, and the determinant  is `r**dimension`, which is `1` if and only if `r` is a root of unity of order `dimension`.

Reply:
There is a typo in `test_PSL()`:

https://github.com/Infleqtion/qLDPC/blob/3ce4ca2f1be931dd5ea9094785ba33819486b602/qldpc/abstract_test.py#L185

The `2` should be `dimension`.

Otherwise looks great, thanks!

Reply:
Awesome, that makes sense.  Thanks!

Reply:
Super, I think it's ready. 
I will draft a PR to merge `PSL` to `main` :)

Reply:
I get an error 

> Pull request creation failed. Validation failed: must be a collaborator

Reply:
Hmm... can you maybe clone the repo and make a PR from `michaelontiveros:PSL` into `Infleqtion:main` (similarly to how you did with this PR)?